extrafiles/tooling/CppIncludeGuards.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""
+This tool iterates over all C++ files and replaces tab-based
+indentation with spaces (tabwidth = 4). If used in read-only
+mode it returns a negative exit code if a change is necessary.
+"""
+import nktooling as nkt
+import re
+import os, sys
+
+nkt.setup()
+
+TabWidth = 4
+
+numScannedFiled = 0
+numNoncompliantFiles = 0
+numIllegalFiles = 0
+
+files = nkt.getCXXFiles(includeSources = False)
+
+msgType = "[ERROR]" if nkt.isReadonly() else "[FIXED]"
+
+processPragmaOnce = False
```

Comment:
What is the intention here? We should not allow `#pragma once` in some files and include guards in others. Since we already have include guards, I see little reason in switching to `#pragma once`.

Reply:
`#pragma once` supposedly speed-up preprocessor runs as file does not have to be parsed several times. Hence, I added this feature to test it. I could however not measure any significant impact. 

To make the intent more explicit I renamed the variable to `includePragmaOnce`; it it's set to false, pragmas added by the user are removed for consistency reasons.